### ISSUE TITLE
[Distributed Index] Intermittent empty results in continuous changes feed

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/sync_gateway#1440]
Using "Couchbase Sync Gateway/1.2.0(35;a2a9c49)", I am getting a doc that looks like this roughly 1 in 5000

```
{"changes": [{"rev": ""}], "id": "", "seq": 10918-0}
```

Full logs (sg1 and sg2 - writers, sg3 - reader) - https://s3.amazonaws.com/sync-gateway-logs/functional_tests-test_continuous.py-test_continuous_changes_parametrized%5BDI-1%5D-2016-01-12-13-23-43-sglogs.zip

This is reproducible in the functional_tests/test_continuous.py::test_continuous_changes_parametrized["DI-1"]

**Comment by sethrosetter** [comment]
More verbose logs - https://s3.amazonaws.com/sync-gateway-logs/functional_tests-test_continuous.py-test_continuous_changes_parametrized%5BDI-1%5D-2016-01-12-14-29-31-sglogs.zip

**Comment by adamcfraser** [comment]
This looks like a BulkGet retry problem to me - where a failed get isn't being retried - but needs some additional diagnostics on the bulk get unmarshal error to be sure.

**Comment by sethrosetter** [comment]
Logs with additional diagnostics - https://s3.amazonaws.com/sync-gateway-logs/functional_tests-test_continuous.py-test_continuous_changes_parametrized%5BDI-1%5D-2016-01-12-15-14-42-sglogs.zip

**Comment by adamcfraser** [comment]
This is failing due to a "Key not found" error when trying to retrieve the entry doc:

```
2016-01-13T00:10:29.521Z WARNING: Non-recoverable error during processGetBatch for key _idx_entry:679:10 :Key not found. -- base.CouchbaseBucketGoCB.processGetBatch() at bucket_gocb.go:405
2016-01-13T00:10:29.521Z WARNING: Error unmarshalling entry for key [_idx_entry:679:10]:unexpected end of JSON input -- db.(*BitFlagStorage).bulkLoadEntries() at kv_channel_storage.go:429
```

The entry document exists in the DB at the end of the test, so it doesn't look like it's just a write error on the index_writer side.

**Comment by adamcfraser** [comment]
Need to review whether this is a case of the reader attempting to process docs that are in the channel clock, but beyond the stable sequence (and so the index doc isn't guaranteed to be written).

**Comment by adamcfraser** [comment]
Looks like the same issue as #1379.

**Comment by adamcfraser** [comment]
This turns out to be a two-part issue.  The more common case is fixed by PR #1467,

There's a second issue that will be tracked by #1465.
